tests/api.c

```diff
@@ -33265,7 +33265,753 @@ static void test_wolfSSL_EVP_PKEY_assign(void)
+        #endif
+        #ifdef WOLFSSL_AES_256
+            NID_aes_256_gcm,
+        #endif
+    #endif /* HAVE_AESGCM */
+    #endif /* (HAVE_FIPS && !HAVE_SELFTEST) || HAVE_FIPS_VERSION > 2 */
+    #ifdef WOLFSSL_AES_COUNTER
+    #ifdef WOLFSSL_AES_128
+         NID_aes_128_ctr,
+    #endif
+    #ifdef WOLFSSL_AES_192
+        NID_aes_192_ctr,
+    #endif
+    #ifdef WOLFSSL_AES_256
+        NID_aes_256_ctr,
+    #endif
+    #endif
+    #ifndef NO_DES3
+         NID_des_cbc,
+         NID_des_ede3_cbc,
+    #endif
+    #ifdef HAVE_IDEA
+         NID_idea_cbc, 
+    #endif
+    };
+    int iv_lengths[] = {
+    #ifdef HAVE_AES_CBC
+            AES_BLOCK_SIZE,
+    #endif
+    #ifdef WOLFSSL_AES_192
+            AES_BLOCK_SIZE,
+    #endif
+    #ifdef WOLFSSL_AES_256
+            AES_BLOCK_SIZE,
+    #endif
+    #if (!defined(HAVE_FIPS) && !defined(HAVE_SELFTEST)) || \
+        (defined(HAVE_FIPS_VERSION) && (HAVE_FIPS_VERSION > 2))
+    #ifdef HAVE_AESGCM
+        #ifdef WOLFSSL_AES_128
+            GCM_NONCE_MID_SZ,
+        #endif
+        #ifdef WOLFSSL_AES_192
+            GCM_NONCE_MID_SZ,
+        #endif
+        #ifdef WOLFSSL_AES_256
+            GCM_NONCE_MID_SZ,
+        #endif
+    #endif /* HAVE_AESGCM */
+    #endif /* (HAVE_FIPS && !HAVE_SELFTEST) || HAVE_FIPS_VERSION > 2 */
+    #ifdef WOLFSSL_AES_COUNTER
+    #ifdef WOLFSSL_AES_128
+            AES_BLOCK_SIZE,
+    #endif
+    #ifdef WOLFSSL_AES_192
+            AES_BLOCK_SIZE,
+    #endif
+    #ifdef WOLFSSL_AES_256
+            AES_BLOCK_SIZE,
+    #endif
+    #endif
+    #ifndef NO_DES3
+            DES_BLOCK_SIZE,
+            DES_BLOCK_SIZE,
+    #endif
+    #ifdef HAVE_IDEA
+            IDEA_BLOCK_SIZE,
+    #endif
+    };
+
+    printf(testingFmt, "wolfSSL_EVP_CIPHER_iv_length");
+    enumlen = (sizeof(enumArray)/sizeof(int));
+    for(i = 0; i < enumlen; i++)
+    {
+        const EVP_CIPHER *c = wolfSSL_EVP_get_cipherbynid(enumArray[i]);
+        AssertIntEQ(wolfSSL_EVP_CIPHER_iv_length(c), iv_lengths[i]);
+    }
+
+    printf(resultFmt, passed);
+#endif
+}
+static void test_wolfSSL_EVP_SignInit_ex(void)
+{
+#if defined(OPENSSL_ALL)
+    WOLFSSL_EVP_MD_CTX  mdCtx;
+    WOLFSSL_ENGINE*     e = 0;
+    const               EVP_MD* md;
+                        md = "SHA256";
+
+    printf(testingFmt, "wolfSSL_EVP_SignInit_ex");
+
+    wolfSSL_EVP_MD_CTX_init(&mdCtx);
+    AssertIntEQ(wolfSSL_EVP_SignInit_ex(&mdCtx, md, e), WOLFSSL_SUCCESS);
+
+    AssertIntEQ(wolfSSL_EVP_MD_CTX_cleanup(&mdCtx), 1);
+
+    printf(resultFmt, passed);
+#endif
+}
+static void test_wolfSSL_EVP_DigestFinal_ex(void)
+{
+#if defined(OPENSSL_ALL) && !defined(NO_SHA256)
+    WOLFSSL_EVP_MD_CTX  mdCtx;
+    unsigned int        s = 0;
+    unsigned char       md[WC_SHA256_DIGEST_SIZE];
+    unsigned char       md2[WC_SHA256_DIGEST_SIZE];
+
+
+    printf(testingFmt, "wolfSSL_EVP_DigestFinal_ex");
+
+
+    /* Bad Case */
+#if !defined(HAVE_FIPS) || (defined(HAVE_FIPS_VERSION) && (HAVE_FIPS_VERSION>2))
```

Comment:
For consistency, please add a some spaces around the ">" in "HAVE_FIPS_VERSION>2".